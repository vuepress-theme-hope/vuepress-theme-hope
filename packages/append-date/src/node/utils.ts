import { Logger } from "@vuepress/helper";

export const PLUGIN_NAME = "vuepress-plugin-append-date";

export const logger = new Logger(PLUGIN_NAME);
