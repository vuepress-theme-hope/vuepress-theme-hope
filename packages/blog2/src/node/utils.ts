import { Logger } from "vuepress-shared";

import type { App, Page } from "@vuepress/core";
import type { PageMap } from "../shared";

export const logger = new Logger("vuepress-plugin-blog2");

export const getPageMap = (
  filter: (page: Page) => boolean,
  app: App
): PageMap => {
  const pageMap: PageMap = {};

  // initialize pageMap
  Object.keys({
    // make sure root locale exists
    // eslint-disable-next-line @typescript-eslint/naming-convention
    "/": {},
    ...app.options.locales,
  }).forEach((path) => {
    pageMap[path] = [];
  });

  app.pages.filter(filter).forEach((page) => {
    pageMap[page.pathLocale].push(page);
  });

  return pageMap;
};
