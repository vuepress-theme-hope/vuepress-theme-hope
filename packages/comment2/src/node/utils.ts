import { Logger } from "vuepress-shared";

export const logger = new Logger("vuepress-plugin-comment2");
