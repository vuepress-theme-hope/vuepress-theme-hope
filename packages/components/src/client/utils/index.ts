export * from "./iframeAllow.js";
export * from "./getLink.js";
export * from "./meta.js";
export * from "./mse.js";
export * from "./pdf.js";
