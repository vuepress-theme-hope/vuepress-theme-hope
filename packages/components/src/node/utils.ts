import { Logger } from "vuepress-shared";

export const logger = new Logger("vuepress-plugin-components");

export const getIconPrefix = (iconAssetsLink = ""): string => {
  if (
    iconAssetsLink === "fontawesome" ||
    iconAssetsLink.match(/^(?:https:)?\/\/kit\.fontawesome\.com\//)
  )
    return "fas fa-";
  if (
    iconAssetsLink === "iconfont" ||
    iconAssetsLink.match(/^(?:https:)?\/\/at\.alicdn\.com\/t\//)
  )
    return "iconfont icon-";

  return "";
};
