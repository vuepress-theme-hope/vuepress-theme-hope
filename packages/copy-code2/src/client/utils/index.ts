export * from "./copy.js";
export * from "./icons.js";
