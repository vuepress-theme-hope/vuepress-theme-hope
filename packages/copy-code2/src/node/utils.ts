import { Logger } from "vuepress-shared";

export const logger = new Logger("vuepress-plugin-copy-code2");
