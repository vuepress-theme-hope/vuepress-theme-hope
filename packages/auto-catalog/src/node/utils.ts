import { getDirname, path } from "@vuepress/utils";
import { Logger, ensureEndingSlash } from "vuepress-shared/node";

const __dirname = getDirname(import.meta.url);

export const PLUGIN_NAME = "vuepress-plugin-auto-catalog";

export const logger = new Logger(PLUGIN_NAME);

export const CLIENT_FOLDER = ensureEndingSlash(
  path.resolve(__dirname, "../client")
);
