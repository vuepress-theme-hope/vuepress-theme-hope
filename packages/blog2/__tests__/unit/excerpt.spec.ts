import { createBaseApp } from "@vuepress/core";
import { path } from "@vuepress/utils";
import { describe, it, expect } from "vitest";

import { emptyTheme } from "./__fixtures__/theme/empty.js";
import { getPageExcerpt } from "../../src/node/index.js";

describe("Should generate page excerpt correctly", async () => {
  const app = createBaseApp({
    bundler: {} as any,
    source: path.resolve(__dirname, "./__fixtures__/src"),
    theme: emptyTheme,
  });

  await app.init();

  it("Should generate excerpt", () => {
    app.pages.forEach((page) => {
      expect(
        getPageExcerpt(app, page, {
          customElement: () => false,
          excerptLength: 200,
          excerptSeparator: "<!-- more -->",
        })
      ).toMatchSnapshot();
    });
  });
});
