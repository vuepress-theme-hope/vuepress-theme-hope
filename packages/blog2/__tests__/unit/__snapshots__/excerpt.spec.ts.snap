// Vitest Snapshot v1

exports[`Should generate page excerpt correctly > Should generate excerpt 1`] = `""`;

exports[`Should generate page excerpt correctly > Should generate excerpt 2`] = `
"<p>A text with </p>
"
`;

exports[`Should generate page excerpt correctly > Should generate excerpt 3`] = `
"<p>A long code block which is expected to be cut of</p>
<div class=\\"language-javascript line-numbers-mode\\" data-ext=\\"js\\"><pre class=\\"language-javascript\\"><code>const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
const a = 1;
</code></pre><div class=\\"line-numbers\\" aria-hidden=\\"true\\"><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div></div></div>"
`;

exports[`Should generate page excerpt correctly > Should generate excerpt 4`] = `
"<p>Here is <strong>article excerpt</strong>.</p>
<div class=\\"language-javascript line-numbers-mode\\" data-ext=\\"js\\"><pre class=\\"language-javascript\\"><code>const a = 1;
</code></pre><div class=\\"line-numbers\\" aria-hidden=\\"true\\"><div class=\\"line-number\\"></div></div></div>"
`;

exports[`Should generate page excerpt correctly > Should generate excerpt 5`] = `
"<h1> Heading 1</h1>
<h2> Heading 2</h2>
<h3> Heading 3</h3>
<h4> Heading 4</h4>
<h5> Heading 5</h5>
<h6> Heading 6</h6>
<h2> Text</h2>
<p>This sentence has <strong>bold</strong>„ÄÅ<em>italic</em> and <s>delete</s> style text.</p>"
`;

exports[`Should generate page excerpt correctly > Should generate excerpt 6`] = `
"<p>This is a <a href=\\"/demo.html\\" target=\\"blank\\">link</a> referencing other markdown.</p>
<p><img src=\\"/logo.svg\\" alt=\\"logo\\"></p>
<p><img src=\\"/logo.svg\\" alt=\\"logo\\"></p>
<p><img src=\\"https://exmaple.com/logo.svg\\" alt=\\"logo\\"></p>"
`;

exports[`Should generate page excerpt correctly > Should generate excerpt 7`] = `""`;
